src/app/ReadClient.cpp

```diff
@@ -296,6 +305,28 @@ CHIP_ERROR ReadClient::GenerateAttributePathList(AttributePathIBs::Builder & aAt
     return aAttributePathIBsBuilder.GetError();
 }
 
+CHIP_ERROR ReadClient::GenerateDataVersionFilterList(DataVersionFilterIBs::Builder & aDataVersionFilterIBsBuilder,
+                                                     DataVersionFilter * apDataVersionFilterList, size_t aDataVersionFilterListSize)
+{
+    for (size_t index = 0; index < aDataVersionFilterListSize; index++)
+    {
+        VerifyOrReturnError(apDataVersionFilterList[index].IsValidDataVersionFilter(), CHIP_ERROR_INVALID_ARGUMENT);
```

Comment:
yep, since we cannot disable default-constructor to avoid invalid data, i would still use optional<DataVersion> mDataVersion, and do check for mDataVersion, thanks